src/UpgradeRule/PHP/Visitor/RenameClassesVisitor.php

```diff
@@ -71,10 +146,42 @@ protected function handleStringUpdate(Scalar\String_ $stringNode)
                 return;
             }
 
-            // Substitute MyClass::class literal in place of string
             $baseName = $this->logUseStatement($replacement);
+
+            // Show warning if this replacement may be invalid
+            if (isset($this->renameWarnings[$baseName]) && $this->showPrompt) {
+
+                $before = $this->source->getNodeLine($stringNode, 'question');
+                $str = 'Attempting to rename: ' . PHP_EOL . $before . PHP_EOL
+                    . 'Do you want to rename '
+                    . $baseName . " to " . $replacement
+                    . " at {$this->file->getPath()}:{$stringNode->getLine()}? (Y/n)";
+
+                $helper = $this->command->getHelper('question');
+                $question = new ConfirmationQuestion($str, true);
+
+                if (!$helper->ask($this->input, $this->output, $question)) {
+                    return;
+                }
+            }
+
+            // Substitute MyClass::class literal in place of string
             $replacementNode = new Expr\ClassConstFetch(new Name([ $baseName ]), 'class');
             $this->source->replaceNode($stringNode, $replacementNode);
+
+            if (isset($this->renameWarnings[$baseName])) {
+                $this->changeSet->addWarning(
+                    $this->file->getPath(),
+                    $stringNode->getLine(),
+                    "Renaming ambiguous string <error>" . $baseName . "</error> to <info>" . $replacement . "</info>\n"
```

Comment:
The original string is not an error, and the red is bit too aggressive. I would have this be info as well.
```suggestion
                    "Renaming ambiguous string <info>" . $baseName . "</info> to <info>" . $replacement . "</info>\n"
```